### PR TITLE
Pill refactor and bug fixing

### PR DESCRIPTION
**Pull request by dearmochi**
## What Does This PR Do

- **General pill refactor.**
  - **Fixes exploit allowing injection of reagents into pills**
  - `New()` -> `Initialize()`
  - Third-person message upon dissolving pill no longer displays to the user, mimicking `visible_message`
  - Make all pill names improper, with the chemical capitalized
  - Fixes grammar in Haloperidol pill
- **Removes `spawn(5)` before deleting the pill when dissolving it into a reagent container.**
  - Touch up some of the related messages to look better in chat

## Why It's Good For The Game

- The refactor should result in a saner code and more consistent user experience, as well as fixing the injection exploit.
- Prevents potential spamming of messages when dissolving a pill. Fortunately there was no duplication exploit from this since reagents are transferred away.

## Changelog
:cl:
tweak: Pill dissolving message is now only visible to others, not self
fix: Fix an exploit allowing injection of reagents into pills
fix: Fix an issue where pills could briefly be dissolved multiple times visually
spellcheck: Chemical names are now capitalized in pre-made pills, as well as being made improper
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_containers/pill.dm [hunk 1]
```diff
@@ -31,34 +32,27 @@
...
-			O.show_message("<span class='warning'>[user] puts something in [target].</span>", 1)
-		spawn(5)
-			qdel(src)
-
-////////////////////////////////////////////////////////////////////////////////
-/// Pills. END
-////////////////////////////////////////////////////////////////////////////////
+	to_chat(user, "<span class='notice'>You dissolve [src] in [target].</span>")
+	for(var/mob/O in oviewers(2, user))
+		O.show_message("<span class='warning'>[user] puts something in [target].</span>", 1)
```
Comment (farie82) [inline code/modules/reagents/reagent_containers/pill.dm:43]: Would be great if this were part of `visible_message`. A range of who sees it I mean. Though out of scope for this PR probably

File: code/modules/reagents/reagent_containers/pill.dm [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
+ * A swallowable pill. Can be dissolved in reagent containers.
+ */
 /obj/item/reagent_containers/food/pill
 	name = "pill"
-	desc = "a pill."
+	desc = "A pill."
 	icon = 'icons/obj/chemical.dmi'
 	icon_state = null
 	item_state = "pill"
+	container_type = 0
```
Comment (farie82) [inline code/modules/reagents/reagent_containers/pill.dm:12]:
```suggestion
	container_type = NONE
```

**Review by farie82 — CHANGES_REQUESTED** [review]
Small styling nitpick. Other than that it's good

**Review by farie82 — APPROVED** [review]

File: code/modules/reagents/reagent_containers/pill.dm [hunk 3]
```diff
@@ -31,34 +32,27 @@
 	return FALSE
 
 /obj/item/reagent_containers/food/pill/afterattack(obj/target, mob/user, proximity)
-	if(!proximity)
+	if(!proximity || !target.is_open_container())
+		return
+	else if(!target.reagents.total_volume)
+		to_chat(user, "<span class='warning'>[target] has no liquid to dissolve [src] in.</span>")
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/pill.dm:38]: I sometimes wonder if we should just get rid of this. It doesn't seem to server a particularly interesting balance function---I know what they're going to, in terms of realism, but being able to just crack a pill open and empty into an empty beaker doesn't seem like it's going to terribly break anything.

Comment (dearmochi) [reply code/modules/reagents/reagent_containers/pill.dm:38]: You have a point. Sinks are aplenty across the station so it wouldn't be hard to add some water to the beaker to dissolve the pill in. At best this is a mild inconvenience.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**